analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyNullableValueAccess.cs

```diff
@@ -321,6 +321,41 @@ class Arithmetic
     int? MultiplicationAndAssignments2(int? i) => (i = null) * (i = 42) * i.Value; // Compliant
 }
 
+class Comparisons
+{
+    void NullIsNeitherSmallerNorBiggerThanANumber(int? i)
+    {
+        i = null;
+        if (i > 0) _ = i.Value;           // Noncompliant, FP: positive, therefore non-empty
+
+        i = null;
+        if (i < 0) _ = i.Value;           // Noncompliant, FP: negative, therefore non-empty
+
+        i = null;
+        if (i == 42) _ = i.Value;         // Compliant, 42, therefore non-empty
+
+        i = Unknown();
+        if (i >= 0 || i < 0)
+            _ = i.Value;                  // Compliant, non-empty
+        else
+            _ = i.Value;                  // FN, empty
+
+        i = Unknown();
```

Comment:
The last two test cases seem redundant to me. They are all doing the same numeric comparison, and from the SE engine's perspective, they look like the same scenario.

Reply:
Removed the second-last test case: indeed, it is very similar to the third-last and whatever solution would be implemented in the SE engine to support one, it would very likely end up supporting the other one too.